sdkproject/Assets/Mapbox/Unity/DataContainers/TileJsonData.cs

```diff
@@ -0,0 +1,191 @@
+	using UnityEngine;
+
+	[Serializable]
+	public class TileJsonData
+	{
+		public readonly string commonLayersKey = "(layer found in more than one source)";
+		public readonly string optionalPropertiesString = "(may not appear across all locations)";
+		/// <summary>
+		/// This boolean is to check if tile JSON data has loaded after the data source has changed
+		/// </summary>
+		public bool tileJSONLoaded = false;
+		/// <summary>
+		/// Layer Display Names seen in the editor
+		/// </summary>
+		public List<string> LayerDisplayNames = new List<string>();
+
+
+		/// <summary>
+		/// Property Display Names seen in the editor
+		/// </summary>
+		public Dictionary<string, List<string>> PropertyDisplayNames = new Dictionary<string, List<string>>();
+
+		/// <summary>
+		/// The description of the property in a layer
+		/// </summary>
+		public Dictionary<string, Dictionary<string, string>> LayerPropertyDescriptionDictionary = new Dictionary<string, Dictionary<string, string>>();
+
+		/// <summary>
+		/// List of data sources (tileset ids) linked to a layer name
+		/// </summary>
+		public Dictionary<string, List<string>> LayerSourcesDictionary = new Dictionary<string, List<string>>();
+
+		/// <summary>
+		/// Dictionary containting the list of layers in a source
+		/// </summary>
+		public Dictionary<string, List<string>> SourceLayersDictionary = new Dictionary<string, List<string>>();
+
+		public void ClearData()
+		{
+			tileJSONLoaded = false;
+			LayerPropertyDescriptionDictionary.Clear();
+			LayerSourcesDictionary.Clear();
+			SourceLayersDictionary.Clear();
+			LayerDisplayNames.Clear();
+			PropertyDisplayNames.Clear();
+		}
+
+		public void ProcessTileJSONData(TileJSONResponse tjr)
+		{
+			List<string> layerPropertiesList = new List<string>();
+			List<string> sourceLayersList = new List<string>();
+
+			if (tjr == null || tjr.VectorLayers == null || tjr.VectorLayers.Length == 0)
+			{
+				return;
+			}
+
+			ClearData();
+
+			var propertyName = "";
+			var propertyDescription = "";
+			var layerSource = "";
+
+			foreach (var layer in tjr.VectorLayers)
+			{
+				//load layer names
+				var layerName = layer.Id;
+				layerPropertiesList = new List<string>();
+				layerSource = layer.Source;
+				if (layer.Fields.Count == 0)
+					continue;
+
+				//loading layer sources
+				if (LayerSourcesDictionary.ContainsKey(layerName))
+				{
+					LayerSourcesDictionary[layerName].Add(layerSource);
+				}
+				else
+				{
+					LayerSourcesDictionary.Add(layerName, new List<string>() { layerSource });
+				}
+
+				//loading layers to a data source
+				if (SourceLayersDictionary.ContainsKey(layerSource))
+				{
+					List<string> sourceList = new List<string>();
+					LayerSourcesDictionary.TryGetValue(layerName, out sourceList);
+
+					if (sourceList.Count > 1 && sourceList.Contains(layerSource)) // the current layerName has more than one source
+					{
+						if (SourceLayersDictionary.ContainsKey(commonLayersKey))
```

Comment:
Seems like `if (SourceLayersDictionary.ContainsKey(commonLayersKey))` only really impacts lines 98 and 107 (aka adding something to SourceLayersDictionary), and the other if-statement is repeated twice. Let's pull out the repeated if-statement to avoid this redundancy, and only write it once.